immunopepper/filter.py

```diff
@@ -79,189 +75,167 @@ def _get_exon_junction_dict(metadata_list, strand):
+                               junction_list: list[str]):
+    """
+    Check if the intron defined by vertex_ids is in the user provided list of junctions
+    :param splicegraph: Spladder splice graph
+    :param vertex_ids: list vertex coordinates in splicegraph defining the intron (typically 2 coordinates)
+    :param strand: '+' for direct, '-' for reverse strand
+    :param junction_list: list of junctions to check the intron against
+    :return: np.nan if junction_list is None or if vertex_ids contains a np.nan
+             1 if the intron is in junction_list, 0 if not
+    """
 
-    if np.nan in vertex_id_list or junction_list is None:
+    if np.nan in vertex_ids or junction_list is None:
         return np.nan
 
     vertex_coord_list = splicegraph.vertices
     if strand == '-':
-        vertex_id_list = vertex_id_list[::-1] # in negative strand case, the vertex id list is [5,3] instead of [3,5]
-    for i in range(len(vertex_id_list) - 1):
-        junc_comb = ':'.join([str(vertex_coord_list[1, vertex_id_list[i]]),  str(vertex_coord_list[0, vertex_id_list[i+1]]), strand])
-        if junc_comb in junction_list:
+        vertex_ids = vertex_ids[::-1]  # in negative strand case, the vertex id list is [5,3] instead of [3,5]
+    for i in range(len(vertex_ids) - 1):
+        junction_comb = ':'.join(
+            [str(vertex_coord_list[1, vertex_ids[i]]), str(vertex_coord_list[0, vertex_ids[i + 1]]), strand])
+        if junction_comb in junction_list:
             return 1
     return 0
 
 
+# TODO(dd): this function smells. It takes nicely structured data and massages it into an undecipherable jumble. Why?
+def add_kmer_properties(foreground_dict, output_kmers: list[OutputKmer]):
+    """
+    Update foreground_dict by collapsing the metadata in output_kmers.
 
-
-def add_dict_kmer_forgrd(foregrd_dict, _namedtuple_list):
-    """...
-        From a namedtuple
-        Updates the following dictionnary:
-        keys = unique peptides,
-        values = collapsed metadata, each metadata item is converted to a set. {'output_id' : set(), 'chr' : set()}
-
-        Parameters
-        ----------
-        foregrd_dict:
-        _namedtuple_list:
-
-        """
-    for _namedtuple_kmer in _namedtuple_list:
-        ### Prepare metadata
-        ord_dict_metadata = _namedtuple_kmer._asdict()
+    :param foreground_dict: keys are peptides, values are lists of property sets, containing the collapsed metadata
+          for the given peptide; for example {'MSPHGYKLA' : [{'ENSMUSG00000025902.13'}, {51.43}, {True}, {1.0}]}
+    :param output_kmers: list of `class:OutputKmer` used to update the dictionary
+    """
+    for output_kmer in output_kmers:
+        # Prepare metadata
+        ord_dict_metadata = output_kmer._asdict()
         del ord_dict_metadata['kmer']
 
-        ### aggregate metadata of unique kmers
+        # aggregate metadata of unique kmers
         add_novel_kmer = []
```

Comment:
This can be put into the first if, bit this is minor.

Reply:
No, no, this is a good catch: it sets the right scope for the variable and makes the code easier to parse.